staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -337,5 +351,15 @@ func (s *ServerRunOptions) AddUniversalFlags(fs *pflag.FlagSet) {
 		"This option, if set, represents the maximum amount of grace period the apiserver will wait "+
 		"for active watch request(s) to drain during the graceful server shutdown window.")
 
-	utilfeature.DefaultMutableFeatureGate.AddFlag(fs)
+	s.FeatureGate.DeferErrorsToValidation(true)
+	s.FeatureGate.AddFlag(fs)
+	s.EffectiveVersion.AddFlags(fs, "")
+}
+
+// Complete sets the final emulation version for the feature gate.
+func (s *ServerRunOptions) Complete() error {
+	if s.FeatureGate.Enabled(genericfeatures.EmulationVersion) {
+		return s.FeatureGate.SetEmulationVersion(s.EffectiveVersion.EmulationVersion())
```

Comment:
ideally, this could be done by the cmd / main package for the command, rather than passing mutable interfaces around where random things could grab them out of the server config

xref where we set feature gates today (either in SetFromMap calls in cmd packages from config, or in flag parsing in cmd packages from flag-binding)